### PR TITLE
Support configmap + add serviceAccountName to report

### PR DESCRIPTION
**Pull request by adisos**

### CODE REVIEW

File: pkg/analyzer/scan.go [hunk 1]
```diff
@@ -127,14 +150,40 @@ func parseDeployResource(podSpec v1.PodTemplateSpec, resourceCtx *common.Resourc
...
+		//consider also cases such as "http://<service name>" with default http port
+		//TODO: could also be a case where value is address as a service name without port, since default port may be used
+		return value + ":80", true //add default port for http
+	}
+	if strings.Contains(value, ":") {
+		return value, true
+	}
+	//TODO: could be a service name as address without default port and without prefix of http://
+	//TODO: what about other protocols prefixes? (https?)
+	//TODO: consider only string values containing services names
```
Comment (zivnevo) [inline pkg/analyzer/scan.go:183]: We should also support references to external websites, not just in-cluster services.
Maybe we can identify services by their typical dns entry `<service-name>.<namespace>.svc.cluster.local` (though just `<service-name>.<namespace>` or `<service-name>` are also legal).

File: pkg/controller/engine.go [hunk 2]
```diff
@@ -73,6 +94,13 @@ func parseResouce(obj parsedK8sObjects) ([]common.Resource, []common.Service) {
 			}
 			res.Resource.FilePath = obj.ManifestFilepath
 			links = append(links, res)
+		} else if p.GroupKind == "ConfigMap" {
+			res, err := analyzer.ScanK8sConfigmapObject(p.GroupKind, p.RuntimeObject)
+			if err != nil {
+				zap.S().Errorf("error scanning Configmap object: %v", err)
+				continue
+			}
+			configMaps = append(configMaps, res)
```
Comment (zivnevo) [inline pkg/controller/engine.go:103]: maybe convert if-else chain to a switch statement

File: pkg/controller/engine.go [hunk 3]
```diff
@@ -21,20 +21,40 @@ func Start(args common.InArgs) error {
...
 	for _, o := range dObjs {
-		r, l := parseResouce(o)
+		r, l, c := parseResouce(o)
 		if len(r) != 0 {
 			resources = append(resources, r...)
 		}
 		if len(l) != 0 {
 			links = append(links, l...)
 		}
+		if len(c) != 0 {
```
Comment (zivnevo) [inline pkg/controller/engine.go:33]: is this `if` needed?

File: pkg/analyzer/scan.go [hunk 4]
```diff
@@ -92,6 +98,23 @@ func ScanK8sDeployObject(kind string, objDataBuf []byte) (common.Resource, error
 	return resourceCtx, nil
 }
 
+func ScanK8sConfigmapObject(kind string, objDataBuf []byte) (common.CfgMapData, error) {
+	var cfgmapCtx common.CfgMapData = make(map[string][]string, 0)
+	obj := ParseConfigMap(bytes.NewReader(objDataBuf))
+
+	//parsed object is a map from ConfigMap's full name (namespace/name) to its data values of interest (list of strings)
+	fullName := obj.ObjectMeta.Namespace + "/" + obj.ObjectMeta.Name
+	data := make([]string, 0)
```
Comment (zivnevo) [inline pkg/analyzer/scan.go:107]:
```suggestion
	data := []string{}
```
More consistent with the rest of the code

**Review by zivnevo — CHANGES_REQUESTED** [review]
Only some minor comments

File: pkg/analyzer/scan.go [hunk 5]
```diff
@@ -127,14 +150,40 @@ func parseDeployResource(podSpec v1.PodTemplateSpec, resourceCtx *common.Resourc
...
 			}
 		}
 	}
 	return nil
 }
 
+//identifyAddressValue checks if value is a potential service address (value is originated from deployment's env or configmap values)
+//It returns a string value (if it's a potential address it may be added with default port) and a bool inidcating
+//if this is indeed a data value of interest as a potential address
+//service addressess considered are of the form "[http://]<service name>:<port number>"
```
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/analyzer/scan.go]: done

File: pkg/controller/engine.go (hunk 2 above)
Comment (adisos) [reply pkg/controller/engine.go]: done

File: pkg/controller/engine.go (hunk 3 above)
Comment (adisos) [reply pkg/controller/engine.go]: removed

File: pkg/analyzer/scan.go (hunk 4 above)
Comment (adisos) [reply pkg/analyzer/scan.go]: done

File: pkg/analyzer/scan.go (hunk 1 above)
Comment (adisos) [reply pkg/analyzer/scan.go:183]: Should we open a separate issue for this? Also, external websites are currently not supported in the created links/connections.

Comment (zivnevo) [reply pkg/analyzer/scan.go:183]: yes, we should probably open a new issue for that

**Review by zivnevo — APPROVED** [review]
ok

**Pull request merged**